### PR TITLE
fix: fenced code block is broken depending on input order (fix #1167)

### LINKED ISSUE #1167: Parsing is broken when the input content is wrapped in an opening code fence syntax
## Describe the bug
Parsing is broken in special cases when using a fenced code block syntax in the markdown editor. If an opening code fence and a closing code fence are sequentially entered or pasted as follows, parsing is correct.

current-correct

But parsing is broken if you enter the content first and then an opening code fence and a closing code fence. In particular, there is a problem when parsing if there are blank lines between the contents of a code block.

current-wrong

## To Reproduce
1. Enter the text `foo` in markdown mode.
2. After `foo`, enter a newline twice, followed by `bar`.
3. Enter an opening code fence before `foo`.
4. Enter a closing code fence after the `bar`.

## Cause

When text is changed in the markdown editor, it is parsed in ToastMark. In the case of a fenced code block, **if there is no a closing code fence after an opening code fence, it is recognized as a code block until the end of the document**.

> Reference : https://spec.commonmark.org/0.29/#example-96

### Wrong Parsing

toastmark-wrong

### Correct Parsing (TODO)

toastmark-correct

## Expected behavior

editor-correct

## Desktop (please complete the following information):
 - OS: All
 - Browser All
 - Version 2.3.1

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description

* Related issue #1167 

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: libs/toastmark/src/toastmark.ts [hunk 1]
```diff
@@ -227,13 +228,15 @@ export class ToastMark {
...
+    const isLastChildCodeBlock = lastChild && isCodeBlock(lastChild);
+    const isLastChildList = lastChild && isList(lastChild);
+
     while (
-      root.lastChild &&
-      isList(root.lastChild) &&
-      nextNode &&
-      (nextNode.type === 'list' || nextNode.sourcepos![0][1] >= 2)
+      (isLastChildCodeBlock && nextNode) ||
+      (isLastChildList && nextNode && (nextNode.type === 'list' || nextNode.sourcepos![0][1] >= 2))
```
Comment (seonim-ryu) [inline libs/toastmark/src/toastmark.ts:239]: > 참조 : 펜스 코드 블럭의 경우 닫기 코드 블럭이 없을 때까지는 문서 끝까지 코드 블럭의 컨텐츠로 처리된다. 
> https://spec.commonmark.org/0.29/#example-96

**Review by lja1018 — APPROVED** [review]
[08/27] 리뷰완료
고생하셨습니다!

File: libs/toastmark/src/toastmark.ts (hunk 1 above)
Comment (shiren) [inline libs/toastmark/src/toastmark.ts:239]: isList라는 함수가 있으니 왠지

```js
isList(nextNode)
```
도 될것 같은데 안되나보네요..

**Review by shiren — APPROVED** [review]
LGTM

**Review by js87zz — APPROVED** [review]
완료! 고생하셨습니다~!

**Pull request merged**